build/scroller.d.ts

```diff
@@ -0,0 +1,139 @@
+type Align = "center";
+
+type Anchors = "hidden" | "visible";
+
+type Scrollbar = "hidden" | "visible";
+
+interface ScrollerConfig<Element extends HTMLElement> {
+  align?: Align;
+  /**
+   * TODO: probably remove
+   * @deprecated please use {anchors} as 'hidden'
+   * @see {Anchors}
+   */
+  noAnchors?: boolean;
+  /**
+   * TODO: probably remove
+   * @deprecated please use {scrollbar} as 'hidden'
+   * @see {Scrollbar}
+   */
+  noScrollbar?: boolean;
+  scrollbar?: Scrollbar;
+  anchors?: Anchors;
+  start?: number;
+  startAnimation?: boolean;
+  el: Element;
+  onClick: (event: MouseEvent | TouchEvent) => void;
+  /**
+   * If we don't need to create markup
+   * e.g. react component will render html by itself
+   *
+   * So we just take outer markup instead
+   */
+  useOuterHtml?: boolean;
+}
+
+type ScrollToPoint = number | `${number}` | "start" | "center" | "end";
+
+// TODO: probably remove
+interface State<ScrollableElement extends HTMLElement> {
+  scrolled: number;
+  scrollable: boolean;
+  pointerDown: boolean;
+  scrollbarPointerDown: boolean;
+  mouseScroll: boolean;
+  scrollbarWidth: number;
+  scrollbarFactor: number;
+  pageX: number[];
+  scrolledDiff: number;
+  downEventTS: number;
+  moveEventTS: number;
+  scrollbarDownPageX: number;
+  scrollClickDisabled: boolean;
+  limitLeft: number;
+  limitRight: number;
+  stripWidth: number;
+  swipeDirection: null | "v" | "h";
+  touchX: number;
+  touchY: number;
+  let: number;
+  el: ScrollableElement | null;
+  isAndroid: boolean;
+}
+
+export interface Scroller<ScrollableElement extends HTMLElement> {
+  new (config: ScrollerConfig<ScrollableElement>): Scroller<ScrollableElement>;
+
+  // public
+  scrollTo(point: ScrollToPoint, time?: number);
+  update(config: ScrollerConfig<ScrollableElement>): void;
+
+  // TODO: probably remove
+  // private
+  checkBorderVisibility(): void;
```

Comment:
they are still available on the prototype though so can be accessed if we remove them

Reply:
Do you mean `#` syntax https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields?

Yeah, that would be cool to have

And polyfill is just putting it to the closure

Reply:
But should we keep them for now?

Reply:
Yup, `#` syntax, right. 

I wouldn't expose them just to avoid misunderstanding in the future. They aren't meant to be used outside anyway 🤷

Reply:
Okay, let me remove them and it would look much nicer

Reply:
Done ✅